code/modules/clothing/suits/armor.dm

```diff
@@ -448,7 +448,13 @@
 
 	name = "[K.name_set] plated armor vest"
 	slowdown = K.slowdown_set
-	armor = K.armor_set
+
+	if (islist(armor) || isnull(armor))		//For an explanation see code/modules/clothing/under/accessories.dm#L39 - accessory detach proc							
+		armor = getArmor(arglist(armor))
+	if (islist(armor) || isnull(armor))
+		armor = getArmor(arglist(armor))
```

Comment:
two ifs?

Reply:
Whoops that's supposed to check the kevlar plate and the vest, not just the vest twice. Fixed